src/MessageList/MessageListItem.vala

```diff
@@ -124,11 +130,33 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         }
 
         add (base_grid);
-        show_all ();
 
         destroy.connect (() => {
             loading_cancellable.cancel ();
         });
+
+        var sender = message_info.from;
+        var whitelist = settings.get_strv ("remote-images-whitelist");
+
+        infobar.response.connect ((id) => {
+            if (id == 2) {
+                if (!(sender in whitelist)) {
+                    whitelist += sender;
+                    settings.set_strv ("remote-images-whitelist", whitelist);
+                }
+            }
+            web_view.load_images ();
+            get_message.begin ();
+            infobar.hide ();
+        });
+
+        if (sender in whitelist) {
+            web_view.load_images ();
```

Comment:
Don't need to do this, the webview already correctly loads images for whitelisted emails.